packages/medusa/src/services/line-item.ts

```diff
@@ -182,6 +182,101 @@ class LineItemService extends TransactionBaseService {
     )
   }
 
+  /**
+   * Generate a single or multiple line item without persisting the data into the db.
+   *
+   * NOTE: this currently work with IsolatedProductModule flag
+   *
+   * @param variantData
+   * @param context
+   */
+  async generateLineItem<
+    T = LineItemCreateData | LineItemCreateData[],
+    TResult = T extends Array<LineItemCreateData> ? LineItem[] : LineItem
+  >(variantData: T, context: GenerateLineItemContext = {}): Promise<TResult> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        this.validateGenerateLineItemArguments(variantData, context)
+
+        const data = variantData
+        const resolvedContext = context as GenerateLineItemContext
+        const regionId = resolvedContext.region_id
+
+        const resolvedData = (
+          Array.isArray(data) ? data : [data]
+        ) as LineItemCreateData[]
+
+        const resolvedDataMap = new Map(
+          resolvedData.map((d) => [d.variant_id, d])
+        )
+
+        const variantsToCalculatePricingFor: {
+          variantId: string
+          quantity: number
+        }[] = []
+
+        for (const variant of resolvedData) {
+          const variantResolvedData = resolvedDataMap.get(variant.variant_id)
+          if (
+            resolvedContext.unit_price == null &&
+            variantResolvedData?.unit_price == null
+          ) {
+            variantsToCalculatePricingFor.push({
+              variantId: variant.variant_id,
+              quantity: variantResolvedData!.quantity,
+            })
+          }
+        }
+
+        let variantsPricing = {}
+
+        // TODO: remove this call and receive prices as input
```

Comment:
question: Just stumbled into this one. This will be a blocker no? I believe we are retrieving variants as part of collecting pricing for variants.

Reply:
I don't think this will be a blocker, but maybe the function should accept the variant_id or the variant as part of the input data? like we do in some places to not have to refectch the data that we already have. Also, when calculating the prices it makes more sense that it only takes care of the calculation based on the input data rather than fetching it. wdyt?

Reply:
but in any case, there is a bit more work to do here @fPolic

Reply:
On it :)